src/praisonai-bot/praisonai_bot/daemon/systemd.py

```
@@ -92,6 +92,22 @@ def uninstall() -> Dict[str, Any]:
         return {"ok": False, "error": str(e)}
 
 
+def restart() -> Dict[str, Any]:
+    """Restart the systemd user service (graceful; unit config drives drain)."""
+    if not os.path.exists(_unit_path()):
+        return {"ok": False, "error": "Service not installed"}
+    try:
+        subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)